app/components/task-selection/component.js

```diff
@@ -226,6 +226,22 @@ export default class TaskSelectionComponent extends Component {
     return this._customersAndRecentTasks.value ?? [];
   }
 
+  get projects() {
+    return this.customer?.projects
+      ?.filter((project) => {
+        return this.archived ? true : !project.archived;
+      })
+      .sortBy("name");
+  }
+
+  get tasks() {
+    return this.project?.tasks
+      ?.filter((task) => {
+        return this.archived ? true : !task.archived;
+      })
+      .sortBy("name");
+  }
+
```

Comment:
```suggestion
  get projects() {
    return this.customer?.projects
      ?.filter(filterByArchived)
      .sortBy("name");
  }

  get tasks() {
    return this.project?.tasks
      ?.filter(filterByArchived)
      .sortBy("name");
  }

  filterByArchived(filterable) {
    return this.archived ? true : filterable.archived;
  }
```

if you wanted you could extract the duplicate filter function into a proper function instead.

Reply:
Good points, thanks!